lib/vrl/stdlib/src/to_regex.rs

```diff
@@ -0,0 +1,76 @@
+use regex::Regex;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToRegex;
+
+impl Function for ToRegex {
+    fn identifier(&self) -> &'static str {
+        "to_regex"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::bytes,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "regex",
+                source: "to_regex(s'foo')",
+                result: Ok(r"foo"),
+            }
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        Ok(Box::new(ToRegexFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ToRegexFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ToRegexFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        use Value::*;
+
+        let value = match self.value.resolve(ctx)? {
+            v @ Bytes(_) => Value::Regex::new(v).unwrap(),
```

Comment:
We'll want to avoid `unwrap()` here as this will cause the program to panic and crash. Instead, I'd recommend returning an error like:

```
let regex = Regex::new(v).map_err(|err| format!("could not create regex: {}",err))?
Value::Regex(regex)
```

This will map the regex compile error to turn it into something the VRL compiler can understand.

Reply:
@jszwedko 
Should I write this inside the `match` block? 
What does `Value::Regex(regex)` do?